logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/message/MessageSenderTest.kt

```diff
@@ -0,0 +1,332 @@
+        runTest {
+            //given
+            setupGivenSuccessResults(
+                getMessageById = true,
+                getConversationProtocol = false,
+                getConversationsRecipient = true,
+                prepareRecipientsForNewOutGoingMessage = true,
+                createOutgoingEnvelope = true,
+                sendEnvelope = true,
+                updateMessageStatus = true
+            )
+            //when
+            val result = messageSender.trySendingOutgoingMessageById(ConversationId("testId", "testDomain"), "testId")
+            //then
+            verify(messageRepository)
+                .suspendFunction(messageRepository::updateMessageStatus)
+                .with(eq(MessageEntity.Status.FAILED), anything(), anything())
+                .wasInvoked(Times(1))
+
+            assertIs<Either.Left<Unit>>(result)
+        }
+    }
+
+    @Test
+    fun givenGettingConversationRecipientsFails_WhenSendingOutgoingMessage_ThenReturnFailureAndSetMessageStatusToFailed() {
+        runTest {
+            //given
+            //given
+            setupGivenSuccessResults(
+                getMessageById = true,
+                getConversationProtocol = false,
+                getConversationsRecipient = true,
+                prepareRecipientsForNewOutGoingMessage = true,
+                createOutgoingEnvelope = true,
+                sendEnvelope = true,
+                updateMessageStatus = true
+            )
+            //when
+            val result = messageSender.trySendingOutgoingMessageById(ConversationId("testId", "testDomain"), "testId")
+            //then
+            verify(messageRepository)
+                .suspendFunction(messageRepository::updateMessageStatus)
+                .with(eq(MessageEntity.Status.FAILED), anything(), anything())
+                .wasInvoked(Times(1))
+
+            assertIs<Either.Left<Unit>>(result)
+            //when
+        }
+    }
+
+    @Test
+    fun testing3() {
+        runTest {
+            //given
+            //given
+            setupGivenSuccessResults(
+                getMessageById = true,
+                getConversationProtocol = true,
+                getConversationsRecipient = false,
+                prepareRecipientsForNewOutGoingMessage = true,
+                createOutgoingEnvelope = true,
+                sendEnvelope = true,
+                updateMessageStatus = true
+            )
+            //when
+            val result = messageSender.trySendingOutgoingMessageById(ConversationId("testId", "testDomain"), "testId")
+            //then
+            verify(messageRepository)
+                .suspendFunction(messageRepository::updateMessageStatus)
+                .with(eq(MessageEntity.Status.FAILED), anything(), anything())
+                .wasInvoked(Times(1))
+
+            assertIs<Either.Left<Unit>>(result)
+        }
+    }
```

Comment:
This test seems to be a copy of the one before, just with a different name.

I suggest:

```suggestion
```